### ISSUE TITLE
Image file upload fails when allowCrossOrigin is true -- FF

### ISSUE THREAD

**Issue by skitterm** [scottcheng/cropit#55]
In Firefox (37), you cannot upload an image when the allowCrossOrigin flag is set to true. The image will fail to load (it will end up in the onImageError function). No problems on Chrome or IE, but similar issue on Mac Safari (8.0.4).

**Comment by nathanbertram** [comment]
+1 seeing this as well

**Comment by scottcheng** [comment]
It seems that FF refuses to load the image when the image object has `crossorigin: 'anonymous', hence `onload` never fires. I'm afraid I don't have a solution at the moment.

**Comment by viciu** [comment]
Is there any reason why allowCrossOrigin option was removed? It looks like previously cropit was able to export images, but after commit e443e51 it doesn't work. 

```
DOMException: Failed to execute 'toDataURL' on 'HTMLCanvasElement': Tainted canvases may not be exported.
```

https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image

**Comment by scottcheng** [comment]
It was removed exactly because of this issue -- it failed entirely on FF. However when I tested again just now it is working in Firefox 38 & 39, Chrome 43 and Safari 8.0.6, but not on Opera 12. I'm adding it back.

**Comment by nathanbertram** [comment]
Thanks guys! Glad FF resolved this issue 



—

**Comment by bedeoverend** [comment]
This seems to be broken again on Safari 8.0.7, and 8.0. Using `crossOrigin = 'true'` works fine on Chrome but not on Safari. Not just cropit, checkout [this codepen](http://codepen.io/bedeoverend/pen/aORQzg), which also doesn't work on FF 39. I've also asked on [stackoverflow](http://stackoverflow.com/questions/31643096/why-does-safari-throw-cors-error-when-setting-image-src-to-base64-data) but no answers yet.

**Comment by bedeoverend** [comment]
Take a look at [this answer](http://stackoverflow.com/a/31667722/4383938) to my stackoverflow question. It looks like both Chrome and Safari / Firefox, are going about it in technically correct ways, but for true cross compatibility, as outlined in the answer, when setting the image `src` it should check if its base64 and _only if it is not_ allow `crossOrigin` to be set to `Anonymous`

**Comment by scottcheng** [comment]
@bedeoverend Thanks a lot for clearing it up -- it's very helpful. I fixed the issue in 0.4.1 according to your answer.

**Comment by bedeoverend** [comment]
@scottcheng fantastic! Thanks
